Assets/Character/Scripts/Run.cs

```diff
@@ -0,0 +1,43 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Run : MonoBehaviour {
```

Comment:
Only saying this for Run but this is the same for the other state components(jump roll etc). Because each state is its own component, and all are active at once, all states call their FixedUpdate() every time. I feel that this is not the best approach, as it doesn't readily telegraph what state is active. A side effect of this is that every state has to set a flag saying if its "on", which i feel isn't very elegant.